apps/browser-extension-wallet/src/utils/get-token-list.ts

```diff
@@ -38,6 +44,123 @@ export interface GetTokenListParams {
+
+  if (nftImage) {
+    return nftImage;
+  }
+
+  return getRandomIcon({ id: info.assetId.toString(), size: 30 });
+};
+
+/**
+ * Retrieves the ticker from NFT metadata if it exists and is a string.
+ *
+ * @param metadata The NFT metadata which might contain a ticker.
+ *
+ * @returns The ticker if present and a string; otherwise, undefined.
+ */
+const getTickerFromNftMetadata = (metadata?: Asset.NftMetadata): string | undefined => {
+  const ticker = metadata?.otherProperties?.get('ticker');
+  return typeof ticker === 'string' ? ticker : undefined;
+};
+
+/**
+ * Retrieves the decimal value from NFT metadata properties.
+ *
+ * This function looks for the 'decimals' property within the NFT metadata's otherProperties map.
+ *
+ * @param metadata - Optional metadata object that may contain the 'decimals' property.
+ *
+ * @returns The decimals value as a number if valid, or 0 if no valid decimals value is found.
+ */
+const getDecimalsFromNftMetadata = (metadata?: Asset.NftMetadata): number => {
+  const rawDecimals = metadata?.otherProperties?.get('decimals');
+
+  if (typeof rawDecimals === 'number') {
+    return rawDecimals;
+  } else if (typeof rawDecimals === 'string' && !Number.isNaN(Number(rawDecimals))) {
+    return Number(rawDecimals);
+  }
+  return 0;
+};
+
+/**
+ * Constructs display metadata for an asset, prioritizing token metadata,
+ * then NFT metadata, and finally using a fallback.
+ *
+ * @param info The asset or handle information containing various metadata.
+ *
+ * @returns An object with name, ticker, decimals and logo of the asset.
+ */
+export const getTokenDisplayMetadata = (
+  info: AssetOrHandleInfo
+): Pick<NonNFTAsset, 'name' | 'description' | 'logo' | 'decimals'> => {
+  const name =
+    info.tokenMetadata?.name ||
+    info.nftMetadata?.name ||
+    addEllipsis(info.fingerprint?.toString() || DISPLAY_FALLBACK, NAME_TRUNCATE_LENGTH, NAME_TRUNCATE_TRAILING_CHARS);
+
+  const ticker = info.tokenMetadata?.ticker || getTickerFromNftMetadata(info.nftMetadata) || DISPLAY_FALLBACK;
+  const decimals = info.tokenMetadata?.decimals || getDecimalsFromNftMetadata(info.nftMetadata);
+
+  return {
+    name,
+    description: ticker,
+    decimals,
+    logo: getTokenLogoUrl(info)
+  };
+};
+
+/**
+ * Retrieves display metadata for an NFT, using available NFT or token metadata.
+ * Falls back to a generated name incorporating the provided chain environment name if no metadata name is found.
+ *
+ * @param info - The asset information containing metadata for NFTs or tokens.
+ * @param environmentName - The name of the blockchain environment, used in default naming.
+ *
+ * @returns An object containing the name and image URL for the NFT.
+ */
+export const getNftDisplayMetadata = (
+  info: AssetOrHandleInfo,
+  environmentName: Wallet.ChainName
+): Pick<NFT, 'name' | 'image'> => {
+  const name = info.nftMetadata?.name || info.tokenMetadata?.name || `SingleNFT${environmentName || ''}`;
```

Comment:
`SingleNFTMainnet` is gibberish
```suggestion
  const name = info.nftMetadata?.name || info.tokenMetadata?.name || tryEncodeToUtf8(info.assetName) || "Unnamed"`;
```

Reply:
@mkazlauskas this was in the original implementation, can/will change if everyone agrees tho

Reply:
preemptively done here https://github.com/input-output-hk/lace/pull/1263/commits/41a4749f818e4fce65496b7a61ea36ac826d911d